aerospike-core/src/bin.rs

```diff
@@ -19,6 +19,7 @@ use serde::Serialize;
 use std::convert::From;
 
 /// Container object for a record bin, comprising a name and a value.
+#[derive(Clone)]
 pub struct Bin<'a> {
     /// Bin name
     pub name: &'a str,
```

Comment:
Are you talking about the Clone or the lifetime on the struct?

Reply:
Struct's `name` field. Can be a `String`, and life will be easier ever after.

Reply:
I agree. That will simplify it a lot.

Reply:
To be frank, I'm worried about the CPU cost of all these little conversions, but my impression is we are saturating the io long before the CPU (async itself is very CPU heavy). I hope we don't come to regret this later. On another note, is there an idiomatic way of accepting `&str`, `String` and `&'static str` in all user-facing APIs?

Reply:
Yes there is. This would work:
```rust
fn main() {
    let stat: &'static str = "static";
    accept_all("&str");
    accept_all("String".to_string());
    accept_all(stat);
}

fn accept_all<S: Into<String>>(value: S) {
    println!("{}", value.into());
}
```